index.js

```diff
@@ -282,7 +282,7 @@ is.arraylike = function (value) {
   return !!value && !is.bool(value)
     && owns.call(value, 'length')
     && isFinite(value.length)
-    && is.number(value.length)
+    && is.integer(value.length)
```

Comment:
Thanks for the contribution!

This is a great intuition to spot - but this isn't actually required; the spec coerces non-integers to integers (see https://tc39.github.io/ecma262/#sec-createlistfromarraylike step 3), so `{ length: 1.2, 0: 'a' }` is a perfectly valid arraylike.